packages/contracts/test/DepositFeed.ts

```diff
@@ -0,0 +1,155 @@
+import { expect } from 'chai'
+import { ethers } from 'hardhat'
+import { Contract, Signer, BigNumber } from 'ethers'
+import { applyL1ToL2Alias } from '@eth-optimism/core-utils'
+
+const ZERO_ADDRESS = '0x' + '00'.repeat(20)
+const ZERO_BIGNUMBER = BigNumber.from(0)
+const NON_ZERO_ADDRESS = '0x' + '11'.repeat(20)
+const NON_ZERO_GASLIMIT = BigNumber.from(50_000)
+const NON_ZERO_VALUE = BigNumber.from(100)
+const NON_ZERO_DATA = '0x' + '11'.repeat(42)
+
+const decodeDepositEvent = (
```

Comment:
Doesn't typechain auto-create nice structure for events? o.o

Reply:
> Is declaring a function as a const holding a lambda how the cool kids do it these days? Is the point to avoid polluting the implicit this?

Yes and yes. :)

> Doesn't typechain auto-create nice structure for events? o.o

Good point, I'm using that now, but the custom function is still helpful to make the data from typechain a bit more usable.